server/server_test.go

```diff
@@ -1994,11 +1993,7 @@ func TestMounts(t *testing.T) {
 
 		t.Run(tst.testName, func(t *testing.T) {
 
-			dir, err := ioutil.TempDir("", strings.Map(nameMapper, tst.testName))
-			if err != nil {
-				panic(err)
-			}
-			defer os.RemoveAll(dir) // Cleanup
+			dir := t.TempDir() // t.TempDir() handles cleanup automatically
```

Comment:
Previously we would create temporary directories, in this case in the pattern `strings.Map(nameMapper, tst.testName)`. I wonder if we should be worried about potential collisions between different tests' use of filesystem paths. Maybe it's best to stick to the current approach, unless you happen to know better.

I see `os` has a similar [CreateTemp](https://pkg.go.dev/os#CreateTemp) function.

Reply:
I don't think we have to worry about this based on the `t.TempDir()` implementation. (You can click the [method name](https://pkg.go.dev/testing#T.TempDir) to see the source code.) It uses the test name as the unique path.

https://cs.opensource.google/go/go/+/refs/tags/go1.24.1:src/testing/testing.go;l=1224-1271